subsys/net/lib/coap/coap.c

```diff
@@ -1817,23 +1819,44 @@ struct coap_reply *coap_response_received(
 		return NULL;
 	}
 
+	type = coap_header_get_type(response);
 	id = coap_header_get_id(response);
 	tkl = coap_header_get_token(response, token);
 
+	if (type == COAP_TYPE_ACK || type == COAP_TYPE_RESET) {
+		piggybacked = true;
```

Comment:
A RST is always a empty message and no piggybacked response.

See [RFC 7252, page 23, table 1](https://www.rfc-editor.org/rfc/rfc7252#page-23)

```
+----------+-----+-----+-----+-----+
|          | CON | NON | ACK | RST |
+----------+-----+-----+-----+-----+
| Request  | X   | X   | -   | -   |
| Response | X   | X   | X   | -   |
| Empty    | *   | -   | X   | X   |
+----------+-----+-----+-----+-----+
  Table 1: Usage of Message Types
```